### ISSUE TITLE
Be able to manually set the external protocol handlers (overriding system settings)

### ISSUE THREAD

**Issue by bsclifton** [brave/brave-browser#781]
Carried over from https://github.com/brave/browser-laptop/issues/1381

The user should be able to configure a handler for `mailto:` and other handlers which would (in-browser) have a higher president than the default app for the OS

There are some security implications for being able to override handlers though and I'm not entirely sure this would be a good idea

**Comment by tomlowenthal** [comment]
@bsclifton I think we currently support this; could you double check?

**Comment by bsclifton** [comment]
@tomlowenthal sorry for the long response time here - you can view the handlers and enable a toggle which will prompt you when new handlers are presented. This is all shown via:
brave://settings/handlers

However, you can't manually add them (at least, as far as I'm aware)

**Comment by 4evermaat** [comment]
So has this been added to brave?   I tried to switch it in windows 10 to default to Brave (previously google chrome opened `mailto:` links just fine.)

Since Brave is based on chromium, not sure why clicking on mailto url within brave browser didn't work.

**Comment by fakepop** [comment]
To add to this: if you remove the existing handler (brave://settings/handlers), there doesn't seem to be a way to reconfigure it, whether manually or automatically by allowing "sites to ask to become default handlers for protocols".

**Comment by DarkAlchy** [comment]
I just left Firefox for Brave, but this is utter nonsense.  If I wanted to be forced to use Gmail for my mailto (only extensions for that) or none in Brave I am stuck using Firefox.  In Chrome I can set it to whatever site I want to handle my email (or program), but not in Brave?

**Comment by Meth0d007** [comment]
I cant get Brave to open my mail client regardless what i set the brave://settings/handlers switch to. Chrome does word and open my default email client on those website links.

**Comment by DarkAlchy** [comment]
> I cant get Brave to open my mail client regardless what i set the brave://settings/handlers switch to. Chrome does word and open my default email client on those website links.

I gave up, and resorted to copying the link to manually open my email to insert it there.  Apparently the devs can't fix it, or simply don't care to.  Shame, but I am seeing the same sort of arrogance I first saw with the Mozilla devs years ago, and look at how (Publiclly stated in their blog too) anti-free speech, etc... they became (the reason I came to Brave).  The parallelism is there.

**Comment by Meth0d007** [comment]
> I gave up, and resorted to copying the link to manually open my email to insert it there. Apparently the devs can't fix it, or simply don't care to. Shame, but I am seeing the same sort of arrogance I first saw with the Mozilla devs years ago, and look at how (Publiclly stated in their blog too) anti-free speech, etc... they became (the reason I came to Brave). The parallelism is there.

Yeah sadly i cant do that, we have an web-app that uses some sort of Javascript to open the mailto: link and somehow closes the dialog again (i think). So i cant copy the link and i can just see a flicker of the "open Outlook watever" popup and then im back to square one.

So stopped rolling out Brave in our company and are reverting back to using FireFox.

**Comment by DarkAlchy** [comment]
> > I gave up, and resorted to copying the link to manually open my email to insert it there. Apparently the devs can't fix it, or simply don't care to. Shame, but I am seeing the same sort of arrogance I first saw with the Mozilla devs years ago, and look at how (Publiclly stated in their blog too) anti-free speech, etc... they became (the reason I came to Brave). The parallelism is there.
> 
> Yeah sadly i cant do that, we have an web-app that uses some sort of Javascript to open the mailto: link and somehow closes the dialog again (i think). So i cant copy the link and i can just see a flicker of the "open Outlook watever" popup and then im back to square one.
> 
> So stopped rolling out Brave in our company and are reverting back to using FireFox.

I completely understand.  Mozilla needs to go the way of an authoritarian dinosaur, so I hate to see you going back to them, but if Brave is incapable (regardless if due to malevolent, or ineptness) of getting the job done it is understandable.

**Comment by Meth0d007** [comment]
Well yes we also dont like the way Mozilla moved in the last years but at least the basics work and they have GPOs for all their stuff (yes pointing at you Brave).

**Comment by lawwantsin** [comment]
How is this still an issue after so many years?  LET PEOPLE EDIT THE HANDLERS!!!

**Comment by knynkwl** [comment]
+1

**Comment by myfta** [comment]
Is this still not resolved in version 102.0.5005.61?

**Comment by 4evermaat** [comment]
[test email](mailto:test@email.com)

I thought the integration was successful now (telling Windows 10 to use Brave as default mail handler)

And for those using Firefox, why not consider WaterFox?   You get 99% of Firefox benefit but with greater privacy controls.  1Password isn't fully integrated though.

**Comment by DarkAlchy** [comment]
I tried Waterfox many years ago and I just did not like it.  As far as Brave goes mail-to just does not seem to work so I just gave up on the idea.  I have to copy and paste it into my email to get around it.  I don't email much these days, but it sure would be nice if it worked for when I do.

I just tried this, for example, mailto:ass@gmail.com and it opened gmail.com I checked in settings->apps->default apps to see my email is set as Brave.  I check the brave://settings/handlers to see other stuff and this:
Not allowed to handle protocols
mail.google.com
email
mail.yandex.com
email

**Comment by myfta** [comment]
My brave://settings/handlers only has an entry for 'Calendar'. There does
not seem to be a way of adding a protocol for email at all.

**Comment by DarkAlchy** [comment]
> My brave://settings/handlers only has an entry for 'Calendar'. There does not seem to be a way of adding a protocol for email at all.
None that I could find either.

**Comment by DarkAlchy** [comment]
> My brave://settings/handlers only has an entry for 'Calendar'. There does not seem to be a way of adding a protocol for email at all.
>

None that I could find either.  I think it is a lost cause at this lifecycle stage of Brave.

**Comment by seumas** [comment]
This is still a vital feature.

**Comment by satcomjimmy** [comment]
I would be happy if it used my system settings. I have my client software defined at the Ubuntu OS and it is not an option to point to from Brave, only gmail

**Comment by moJinn** [comment]
Also, some websites need to be handlers for certain types of links not just windows apps, but when I make a website default in (brave://settings/handlers) everything works fine until i restart the browser as the default website for the handler gets reset. When I toggle the website to be the default handler, windows 11 default selector window opens up expecting me to make an app the default... This functionality is broken.  This issue seems to be abandoned. I was loving brave until i came across this issue, and seeing how long this has been going on, it's discouraging. Sadly is a deal breaker for me as well.

**Comment by andyl** [comment]
I'm another one affected by this problem.  Do not want to install Firefox.  Which browser should I use?

**Comment by prl77** [comment]
Same here. Windows already has Brave as default for mail. Clicking a mailto link opens a blank tab. None of the webmail clients I tried have the double diamond icon, including Gmail, Proton, Zoho.
